### PR TITLE
Add failing spec for Elixir resolution in an umbrella app

### PR DESCRIPTION
**Pull request by greysteil**
This combination of mixfiles sends Hex into a tailspin. Running `mix deps.get` on it locally causes my computer to melt.

In Dependabot, we time out when that happens. However, we appear not to be returning anything from the update checker when that happens, causing the following error:

```
** (ArgumentError) argument error
           :erlang.binary_to_term("")
           /Users/greysteil/code/dependabot-core/helpers/elixir/bin/run.exs:11: DependencyHelper.main/0
           (elixir) lib/code.ex:767: Code.require_file/2
           (mix) lib/mix/tasks/run.ex:146: Mix.Tasks.Run.run/5
```

This PR adds a failing spec for the above case, in the hope that @stevedomin might be kind enough to contribute a fix.

### CODE REVIEW

**Comment by greysteil** [comment]
@stevedomin - OK, I think I've boiled this down to the point where it will be 1 minute of your time. Can you take a look? `bundle exec rspec spec/dependabot/update_checkers/elixir/hex_spec.rb:416` to replicate, and you'll notice the spec takes 30 seconds to run, almost certainly because it's timing out.

**Comment by greysteil** [comment]
@stevedomin - I wrote the Elixir to fix. If you could give it a 30 second review that would be amazing, but I'm going to merge this if specs pass anyway.

File: helpers/elixir/bin/check_update.exs [hunk 1]
```diff
@@ -19,7 +19,7 @@ defmodule UpdateChecker do
 
       {:ok, {:error, error}} -> {:error, error}
 
-      nil -> {:error, :dependency_resolution_timed_out}
+      {:ok, nil} -> {:error, :dependency_resolution_timed_out}
```
Comment (stevedomin) [inline helpers/elixir/bin/check_update.exs]: Proper fix is actually on line 7, I shouldn't have removed the chained `Task.shutdown(task)`

```elixir
task = Task.async(fn -> do_resolution(dependency_name) end)
case Task.yield(task, 30000) || Task.shutdown(task) do
  {:ok, {:ok, :resolution_successful}} ->
    ...
  {:ok, {:error, error}} -> {:error, error}

  nil -> {:error, :dependency_resolution_timed_out}

  {:exit, reason} -> {:error, reason}
end
```

Comment (greysteil) [reply helpers/elixir/bin/check_update.exs]: Awesome, thanks!

**Pull request merged**